datafusion/sql/src/relation/mod.rs

```diff
@@ -143,4 +147,45 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
             Ok(plan)
         }
     }
+
+    pub(crate) fn create_relation_subquery(
+        &self,
+        subquery: TableFactor,
+        planner_context: &mut PlannerContext,
+    ) -> Result<LogicalPlan> {
+        // At this point for a syntacitally valid query the outer_from_schema is
+        // guaranteed to be set, so the `.unwrap()` call will never panic. This
+        // is the case because we only call this method for lateral table
+        // factors, and those can never be the first factor in a FROM list. This
+        // means we arrived here through the `for` loop in `plan_from_tables` or
+        // the `for` loop in `plan_table_with_joins`.
+        let old_from_schema = planner_context.set_outer_from_schema(None).unwrap();
+        let new_query_schema = match planner_context.outer_query_schema() {
+            Some(lhs) => Some(Arc::new(lhs.join(&old_from_schema)?)),
```

Comment:
We probably want to replace the `lhs.join(<rhs>)` call here with aversion of a `join` where duplicate columns on the right side override their `lhs` counterpart.

See #11464 for details.

Reply:
I added a second commit to the PR that fixes the issue mentioned above (see the commit message for details).